File: frontend/lib/doctor.ts
import api from "./api";

export interface Doctor {
  id?: number;
  name: string;
  specialization: string;
  qualification: string;
  experience: number;
  email: string;
  phone: string;
  consultation_fee: number;
  available: boolean;
}

export async function getDoctors(
  token: string,
  search = ""
) {
  const response = await api.get("/doctors", {
    params: {
      search,
    },
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  return response.data;
}

export async function createDoctor(
  doctor: Doctor,
  token: string
) {
  const response = await api.post(
    "/doctors",
    doctor,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}

export async function updateDoctor(
  id: number,
  doctor: Doctor,
  token: string
) {
  const response = await api.put(
    `/doctors/${id}`,
    doctor,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}

export async function deleteDoctor(
  id: number,
  token: string
) {
  await api.delete(`/doctors/${id}`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
}

File: backend/app/schemas/doctor.py
from pydantic import BaseModel, ConfigDict, EmailStr


class DoctorCreate(BaseModel):
    name: str
    specialization: str
    qualification: str
    experience: int
    email: EmailStr
    phone: str
    consultation_fee: float
    available: bool = True


class DoctorUpdate(BaseModel):
    name: str | None = None
    specialization: str | None = None
    qualification: str | None = None
    experience: int | None = None
    email: EmailStr | None = None
    phone: str | None = None
    consultation_fee: float | None = None
    available: bool | None = None


class DoctorRead(BaseModel):
    id: int
    name: str
    specialization: str
    qualification: str
    experience: int
    email: EmailStr
    phone: str
    consultation_fee: float
    available: bool

    model_config = ConfigDict(from_attributes=True)